src/main/perl/CacheManager.pm

```diff
@@ -137,6 +145,167 @@ sub getCachePath
     return $self->{"cache_path"};
 }
 
+=item getConfigurationPath
+
+For given C<cid>, return the basepath of the Configuration data.
+(No checks are made e.g. if the directory exists,
+simply returns the directory name).
+
+=cut
+
+sub getConfigurationPath
+{
+    my ($self, $cid) = @_;
+
+    my $cache_path = $self->getCachePath();
+
+    return "$cache_path/$PROFILE_DIR_N$cid";
+}
+
+=item getCids
+
+Return arrayref to sorted list of all found/valid CIDs.
+
+Returns undef in case of problem.
+
+=cut
+
+sub getCids
+{
+    my ($self, $cid) = @_;
+
+    my $cache_path = $self->getCachePath();
+
+    my $cid_pattern = '(?:^|/)' . $PROFILE_DIR_N . '(\d+)$';
+
+    return if( ! opendir(DIR , $cache_path));
+
+    my @cids = sort map { m/$cid_pattern/; $1 } grep { m/$cid_pattern/ && -d "$cache_path/$_" } readdir(DIR);
+    close(DIR);
+
+    return \@cids;
+}
+
+=item getCid
+
+For given C<cid>, validate and check the CID.
+
+Returns undef for a non-existing CID.
+
+Also handles special values for C<cid>:
+
+=over
+
+=item undef, "current" or empty string
+
+If CID is undef, the string "current" or an empty string, the current CID
+(from the "current.cid" file) is returned.
+
+=item "latest" or "-"
+
+If CID is the string "latest" or "-", the latest CID
+(from the "latest.cid" file) is returned.
+
+=item negative value (e.g. -1)
+
+If CID is negative C<-N>, the N-th most recent CID value is returned
+(e.g. -1 returns the most recent CID, -2 the CID before the most recent, ...).
+
+(A distinction is made between "most recent" and "latest", as the "latest" CID
+is held in the "latest.cid" file).
+
+=back
+
+=cut
+
+sub getCid
+{
+    my ($self, $cid) = @_;
+
```

Comment:
`$cids` is an array reference, making the scalar items `scalar @$cids`; but i renamed `$cids` to `$valid_cids` to make it more clear.